src/main/java/nl/novi/serviceconnect/services/TransactionService.java

```diff
@@ -0,0 +1,118 @@
+
+            throw new RuntimeException("Error generating and saving PDF file", e);
+        }
+    }
+
+    @Override
+    public TransactionOutputDto createTransaction(TransactionInputDto transactionInputDto) {
+
+        Transaction transaction = Mapper.fromDtoToTransaction(transactionInputDto);
+
+        String filePath = saveFile();
+        transaction.setInvoice(new File(filePath));
+
+        ServiceRequest serviceRequest = requestRepository.findById(transactionInputDto.getServiceRequestId())
+                .orElseThrow(() -> new RecordNotFoundException("ServiceRequestId not found with id: " + transactionInputDto.getServiceRequestId()));
+        transaction.setServiceRequest(serviceRequest);
+
+       transactionRepository.save(transaction);
+
+        return Mapper.fromTransactionToDto(transaction);
+    }
+
+    @Override
+    public List<TransactionOutputDto> getAllTransaction() {
+        List<Transaction> transactionList = transactionRepository.findAll();
+        List<TransactionOutputDto> transactionDtoList = new ArrayList<>();
+
+        if (transactionList.isEmpty()) {
+            throw new RecordNotFoundException("No Transaction found");
+        }
+
+        for(Transaction transaction : transactionList) {
+            transactionDtoList.add(Mapper.fromTransactionToDto(transaction));
+        }
+        return transactionDtoList;
+    }
+
+    @Override
+    public TransactionOutputDto getTransactionById(Long id) {
+        Optional<Transaction> transaction = transactionRepository.findById(id);
+
+        Transaction result = transaction.orElseThrow(() ->
+                new RecordNotFoundException("No transaction found with id: " + id));
+
+        return Mapper.fromTransactionToDto(result);
+    }
+
+    @Override
+    public TransactionOutputDto updateTransaction(Long id, TransactionInputDto inputDto) {
+        Transaction existingTransaction = transactionRepository.findById(id)
+                .orElseThrow(() -> new RecordNotFoundException("Transaction with id: " + id + " not found"));
+        Transaction result =  updateServiceFields(existingTransaction, inputDto);
+        transactionRepository.save(result);
+
+        return Mapper.fromTransactionToDto(result);
+    }
+
+    private Transaction updateServiceFields(Transaction transaction, TransactionInputDto inputDto) {
+
+        return new Transaction(
+                transaction.getId(),
+                Helpers.isNotNullOrEmptyDate(inputDto.getTransactionDate()) ? inputDto.getTransactionDate() : transaction.getTransactionDate(),
+                inputDto.getPayed()
+        );
+    }
+    @Override
+    public void deleteTransaction(Long id) {
+        Optional<Transaction> transaction = transactionRepository.findById(id);
+
+        if (transaction.isPresent()) {
+            transactionRepository.deleteById(id);
+        } else {
+            throw new RecordNotFoundException("Transaction with id " + id + " not found");
+        }
```

Comment:
Het is een betere optie om de unhappy paden aan het begin te hebben. Op die manier kun je ook je nesting weer verminderen.
```
if (!transaction.IsPresent()) throw new ... ;

transactionRepository.deleteById();
```

Reply:
ga ik zeker doen thx

Reply:
ik heb het zo opgelost

 if (transaction.isEmpty()) {
            throw new RecordNotFoundException("Transaction with id " + id + " not found");
        } else {
            transactionRepository.deleteById(id);
        }